### PR TITLE
UPSTREAM: google/cadvisor: 1963: container: fix concurrent map acccess

### PR DESCRIPTION
**Pull request by runcom**
Signed-off-by: Antonio Murdaca <runcom@redhat.com>

@sjenning PTAL, picks https://github.com/google/cadvisor/pull/1963 for https://bugzilla.redhat.com/show_bug.cgi?id=1590563

/cc @deads2k

### CODE REVIEW

**Comment by sjenning** [comment]
merged upstream and lgtm

**Comment by liggitt** [comment]
this got pulled into dep update in https://github.com/openshift/origin/pull/19992 and doesn't appear to compile... is there a coordinating PR required?

```
# github.com/openshift/origin/vendor/github.com/google/cadvisor/container/docker
vendor/github.com/google/cadvisor/container/docker/handler.go:250:9: handler.restartCount undefined (type *dockerContainerHandler has no field or method restartCount)
[ERROR] [18:10:08+0000] PID 78: hack/lib/build/binaries.sh:236: `GOOS=${platform%/*} GOARCH=${platform##*/} go install -pkgdir "${pkgdir}/${platform}" -tags "${OS_GOFLAGS_TAGS-} ${!platform_gotags_envvar:-}" -ldflags="${local_ldflags}" "${goflags[@]:+${goflags[@]}}" -gcflags "${gogcflags}" "${nonstatics[@]}"` exited with status 2.
```

**Comment by deads2k** [comment]
> this got pulled into dep update in openshift/origin#19992 and doesn't appear to compile... is there a coordinating PR required?

update glide.yaml to point to a SHA and they'll deal with it.

File: container/docker/handler.go [hunk 1]
```diff
@@ -251,6 +248,10 @@ func newDockerContainerHandler(
 	handler.networkMode = ctnr.HostConfig.NetworkMode
 	handler.deviceID = ctnr.GraphDriver.Data["DeviceId"]
 	handler.restartCount = ctnr.RestartCount
```
Comment (sjenning) [inline container/docker/handler.go:250]: this line needed to be removed

Comment (sjenning) [reply container/docker/handler.go:250]: it is correct upstream :-/  not sure how it didn't make the pick.

Comment (runcom) [reply container/docker/handler.go:250]: I'm not sure either indeed, my bad guys

**Pull request merged**